gtsfm/scene_optimizer.py

```diff
@@ -336,6 +363,50 @@ def write_sfmdata_to_disk(sfm_data: SfmData, save_fpath: str) -> None:
     gtsam.writeBAL(save_fpath, sfm_data)
 
 
+def aggregate_frontend_metrics(
+    rot3_errors: List[float],
+    unit3_errors: List[float],
+    angular_err_threshold: float,
+) -> None:
+    """Aggregate the front-end metrics to log summary statistics.
+
+    Args:
+        rot3_errors: angular errors in rotations.
+        unit3_errors: angular errors in unit-translations.
+    """
+    # TODO: fix the types
+    num_entries = len(rot3_errors)
+
+    rot3_errors = np.array(rot3_errors)
+    unit3_errors = np.array(unit3_errors)
+
+    # compute pose errors by picking the max error
+    pose_errors = np.maximum(rot3_errors, unit3_errors)
+
+    # check errors against the threshold
+    success_count_rot3 = np.sum(rot3_errors < angular_err_threshold)
+    success_count_unit3 = np.sum(unit3_errors < angular_err_threshold)
+    success_count_pose = np.sum(pose_errors < angular_err_threshold)
+
+    logging.debug(
+        "[Two view optimizer] [Summary] Rotation success: %d/%d",
+        success_count_rot3,
+        num_entries,
+    )
+
+    logging.debug(
```

Comment:
I would prefer `logger.info` instead of `logging.debug`

Reply:
It doesn't work with `logger.info`. I might be doing something wrong here

Reply:
As discussed, leaving it for later